Doc/library/threading.rst

```diff
@@ -594,14 +594,24 @@ and "recursion level" in addition to the locked/unlocked state used by primitive
 locks.  In the locked state, some thread owns the lock; in the unlocked state,
 no thread owns it.
 
-To lock the lock, a thread calls its :meth:`~RLock.acquire` method; this
-returns once the thread owns the lock.  To unlock the lock, a thread calls
-its :meth:`~Lock.release` method. :meth:`~Lock.acquire`/:meth:`~Lock.release`
-call pairs may be nested; only the final :meth:`~Lock.release` (the
-:meth:`~Lock.release` of the outermost pair) resets the lock to unlocked and
-allows another thread blocked in :meth:`~Lock.acquire` to proceed.
+Threads call a lock's :meth:`~RLock.acquire` method to lock it,
+and its :meth:`~Lock.release` method to unlock it. Reentrant locks
+support the :ref:`context management protocol <with-locks>` so :keyword:`with` can be used
+to acquire and release the lock automatically in a block of code.
 
-Reentrant locks also support the :ref:`context management protocol <with-locks>`.
+.. seealso::
+
+   :ref:`with-locks`.
```

Comment:
```suggestion
   :ref:`Using RLock as a context manager <with-locks>` is recommended
   over manual :meth:`!acquire` and :meth:`release` calls whenever practical.
```

Here's how I would phrase this, and also move it down past the first summary sentence in `RLock.acquire()` ("Acquire a lock, blocking or non-blocking")